Update tini install snippet (#5408)
docker-library/official-images: README.md

Following the Docker guidelines it is highly recommended that the resulting image be just one concern per container; predominantly this means just one process per container, so there is no need for a full init system. There are two situations where an init-like process would be helpful for the container. The first being signal handling. If the process launched does not handle `SIGTERM` by exiting, it will not be killed since it is PID 1 in the container (see "NOTE" at the end of the [Foreground section](https://docs.docker.com/engine/reference/run/#foreground) in the docker docs). The second situation would be zombie reaping. If the process spawns child processes and does not properly reap them it will lead to a full process table, which can prevent the whole system from spawning any new processes. For both of these concerns we recommend [tini](https://github.com/krallin/tini). It is incredibly small, has minimal external dependencies, fills each of these roles, and does only the necessary parts of reaping and signal forwarding.

Be sure to use tini in `CMD` or `ENTRYPOINT` as appropriate.

> It is best to install tini from a package. If tini is not available as a package here is a snippet of a `Dockerfile` to add in tini:

Reviewer: ```suggestion
It is best to install tini from a distribution-provided package (ex. `apt-get install tini`). If tini is not available in your distribution or is too old, here is a snippet of a `Dockerfile` to add in tini:
```

```Dockerfile

> # alpine:3.4+ rather use: apk add --no-cache tini

Reviewer: Rather than maintaining a list of which distributions have `tini` available, we should just stick to the "get it as a package".
```suggestion
```

> # ubuntu:18.10+ rather use: apt-get install -y tini

Reviewer: ```suggestion
```

> # debian:10+ rather use: apt-get install -y tini

Reviewer: ```suggestion
```

> #

Reviewer: ```suggestion
```

# Install tini for signal processing and zombie killing
ENV TINI_VERSION v0.18.0
ENV TINI_SIGN_KEY 595E85A6B1B4779EA4DAAEC70B588DFF0527A9B7
RUN set -eux; \
  wget -O /usr/local/bin/tini "https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini"; \
  wget -O /usr/local/bin/tini.asc "https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini.asc"; \
  export GNUPGHOME="$(mktemp -d)"; \

>   gpg --batch --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys "$TINI_SIGN_KEY" || \

Reviewer: This is just, the loop written out; for official-images we only require one server (`ha.pool.sks-keyservers.net`), since we run a proxy to ensure it fails less often. It is basically https://github.com/docker-library/php/pull/666.
```suggestion
```

>   gpg --batch --keyserver hkp://ipv4.pool.sks-keyservers.net --recv-keys "$TINI_SIGN_KEY" || \

Reviewer: ```suggestion
```

>   gpg --batch --keyserver hkp://pgp.mit.edu:80 --recv-keys "$TINI_SIGN_KEY" || \

Reviewer: ```suggestion
```

```Dockerfile
# grab tini for signal processing and zombie killing
ENV TINI_VERSION v0.18.0

> RUN set -ex \

Reviewer 2: We should probably go all the way if we're updating this and do `set -eux;` and use semicolons instead since their `set -e` behavior is much more reliable (since `&&` creates a "complex expression" and the `set -e` rules become less predictable, especially with things like `for` loops embedded).

	&& curl -fSL --compressed "https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-static" -o /usr/local/bin/tini \

> 	&& curl -fSL --compressed "https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-static.asc" -o /usr/local/bin/tini.asc \

Reviewer 2: In Alpine, `tini` has been available as a package since Alpine 3.4, so I don't think it's very prudent for us to show how to install a static version just for Alpine's sake (since `apk add --no-cache tini` is going to be superior there in almost every way except the ability to get the latest bleeding edge, which is totally sane as an exercise for the reader).

Reviewer 2: Why does this include `--compressed` now?  IMO we should instead just switch to `wget` which is easier to get right out-of-the-box (no `-f` needed, for example):

```bash
wget -O /usr/local/bin/tini "https://...."; \
```

	&& export GNUPGHOME="$(mktemp -d)" \
	&& for key in \
				595E85A6B1B4779EA4DAAEC70B588DFF0527A9B7 ; \
			do \
				gpg --batch --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys "$key" || \
				gpg --batch --keyserver hkp://ipv4.pool.sks-keyservers.net --recv-keys "$key" || \
				gpg --batch --keyserver hkp://pgp.mit.edu:80 --recv-keys "$key" || \

> 				gpg --batch --keyserver ha.pool.sks-keyservers.net --recv-keys "$key" ; \

Reviewer 2: This keyserver looping (and looping over a single key) is unnecessary for the official images build infrastructure and definitely not something we'd recommend (see https://github.com/docker-library/official-images/issues/4252#issuecomment-405749749, https://github.com/docker-library/php/pull/666, https://github.com/tianon/pgp-happy-eyeballs).

			done \
	&& gpg --batch --verify /usr/local/bin/tini.asc /usr/local/bin/tini \

> 	&& rm -rf "${GNUPGHOME}" /usr/local/bin/tini.asc \

Reviewer 2: I'm not sure the `{}` add any value here?  Also, we should probably include `gpgconf --kill all` before this line (or `command -v gpgconf && gpgconf --kill all || :`, if we want to be hyper-explicit).
